### PR TITLE
Fix small graphite round trip bugs

### PR DESCRIPTION
**Pull request by mhosken**
This patch fixes a few minor but significant bugs in the Graphite table support to ensure clean round tripping via ttx.

### CODE REVIEW

File: Lib/fontTools/ttLib/tables/S__i_l_f.py [hunk 1]
```diff
@@ -727,18 +727,18 @@ def decompile(self, data, ttFont, version=2.0):
                 oConstraints[i] = oConstraints[i+1]
         self.ruleConstraints = [(data[s:e] if (e-s > 1) else "") for (s,e) in zip(oConstraints, oConstraints[1:])]
         data = data[oConstraints[-1]:]
-        for i in range(len(oActions)-2,-1,-1):
-            if oActions[i] == 0:
-                oActions[i] = oActions[i+1]
+#        for i in range(len(oActions)-2,-1,-1):
+#            if oActions[i] == 0:
+#                oActions[i] = oActions[i+1]
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/tables/S__i_l_f.py]: why you commented this out instead of simply deleting the lines?

**Comment by anthrotype** [comment]
the test data also needs to be adjusted as you see from the bots failure.

**Comment by anthrotype** [comment]
we should probably handle the case when TTX file was generated with a previous version of fonttools and does not have that new "default" attribute

**Comment by anthrotype** [comment]
for py2.py3 compatiblity here you should use `byteord` instead of plain `ord`:
https://travis-ci.org/fonttools/fonttools/jobs/296902189#L2937

on python 3 indexing a bytes string yields an int, not a character.

for this other test failure, you need to convert the iterator returned by `map` function to a list if you want to get it's `len()`
https://travis-ci.org/fonttools/fonttools/jobs/296902189#L2992-L2993

**Comment by anthrotype** [comment]
`byteord` is our compatibility wrapper defined in [`fontTools.misc.py23`](https://github.com/fonttools/fonttools/blob/b9bca702f6efb06e9c3c77f86bbb824d84b7b810/Lib/fontTools/misc/py23.py#L104). If you did `from fontTools.misc.py23 import *` at the top, then it is already in your namespace, otherwise you can import explicitly `from fontTools.misc.py23 import byteord`

**Comment by mhosken** [comment]
Thanks for the tip on byteord. Would probably have reinvented the wheel, badly, wasted time and resulted in not good code.

**Pull request merged**